source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -279,36 +290,31 @@ namespace aspect
       }
 
 
-
       template <int dim>
       std::pair<double, double>
-      CompositeViscoPlastic<dim>::compute_strain_rate_and_derivative (const double creep_stress,
-                                                                      const double pressure,
-                                                                      const double temperature,
-                                                                      const DiffusionCreepParameters diffusion_creep_parameters,
-                                                                      const DislocationCreepParameters dislocation_creep_parameters,
-                                                                      const PeierlsCreepParameters peierls_creep_parameters,
-                                                                      const DruckerPragerParameters drucker_prager_parameters) const
+      CompositeViscoPlastic<dim>::calculate_log_strain_rate_and_deriv(const std::vector<std::pair<double, double>> &log_edot_and_deriv) const
```

Comment:
I know this may propagate eventually through all flow laws, but I always thought these variable names need some improvement. What about:
```suggestion
      CompositeViscoPlastic<dim>::calculate_log_strain_rate_and_deriv(const std::vector<std::pair<double, double>> &logarithmic_strain_rates_and_stress_derivatives) const
```
This clarifies which derivative is handed over, and it also clarifies that there are several in that vector, not just one. Plus everyone who is not familiar with the nomenclature in the equations understands the name.